examples/login-once-per-worker/config.ts

```diff
@@ -14,14 +14,26 @@
  * limitations under the License.
  */
 
-import { folio as baseFolio } from '@playwright/test';
+import { BrowserContext } from 'playwright';
+import { ChromiumEnv, FirefoxEnv, WebKitEnv, PlaywrightOptions, setConfig, newTestType, merge } from '@playwright/test';
+
+// Global Folio configuration.
+setConfig({
+  testDir: __dirname,
+  timeout: 30000,
+});
+
+// New test type that provides a number in addition to standard page, context, etc.
+export const test = newTestType<{ loginOnce: (context: BrowserContext) => Promise<BrowserContext> }>();
 export { expect } from '@playwright/test';
 
-const builder = baseFolio.extend<{}, { loggedInState: any }>();
+let loggedInState: any;
```

Comment:
instead of having the state here in a global variable, I would move it in an instance variable of LoggedInEnv. I would move the functions also directly into this as a private methods.

Reply:
Yes I tried to place the instance into `LoggedInEnv`, but the error `this.loggedInState is not defined` occurred.

Maybe `merge` function extracts only the specific property like this:

```ts
const { beforeEach } = LoggedInEnv;
```

Reply:
I played a bit around, seems like that would work: https://gist.github.com/mxschmitt/85c18efe336d9c7307f10046459a4023

Reply:
Yes my expectation is more or less correct:

https://github.com/microsoft/folio/blob/2202f43ba956bed88c1b78464351ef40ed81b1e0/src/spec.ts#L47-L98

Reply:
yeah agree, that's super hacky. Will try to follow up on folio or you if you want to fix it, feel free to open a PR there.

Reply:
Your solution works perfectly!

I fixed this in https://github.com/microsoft/playwright-test/pull/246/commits/bde971daeea61a880a5472c110efb70f726424e6